gui/src/avopenglwidget.cpp

```diff
@@ -256,26 +309,31 @@ void AVOpenGLWidget::initializeGL()
 
 	for(int i=0; i<2; i++)
 	{
-		f->glGenTextures(3, frames[i].tex);
-		f->glGenBuffers(3, frames[i].pbo);
-		uint8_t uv_default = 127;
-		for(int j=0; j<3; j++)
+		frames[i].conversion_config = conversion_config;
+		f->glGenTextures(conversion_config->planes, frames[i].tex);
+		f->glGenBuffers(conversion_config->planes, frames[i].pbo);
+		uint32_t uv_default = 0x7f7f7f7f;
```

Comment:
I don't think this is right for nv12, it shows up as dark green. Since it's more than one byte now, it should be better to do something like this to not run into endianness issues:
```c++
uint8_t uv_default[] = [0x7f, 0x7f];
```

Reply:
So funny fact - it worked before my rework. By then I actually assigned 127 << 8 | 127, what is effectively the same. I have no idea why it went dark-green.

Do you want me to iterate on it or shall we ignore it for now and just iterate?